napari/layers/points/points.py

```diff
@@ -380,6 +588,126 @@ def edge_width(self, edge_width: Union[None, float]) -> None:
         self.events.edge_width()
         self.events.highlight()
 
+    @property
+    def edge_color(self):
+        """(N x 4) np.ndarray: Array of RGBA edge colors for each point"""
+        return self._edge_color
+
+    @edge_color.setter
+    def edge_color(self, edge_color):
+        # if the provided face color is a string, first check if it is a key in the properties.
+        # otherwise, assume it is the name of a color
+        if self._is_color_mapped(edge_color):
+            if guess_continuous(self.properties[edge_color]):
+                self._edge_color_mode = ColorMode.COLORMAP
+            else:
+                self._edge_color_mode = ColorMode.CYCLE
+            self._edge_color_property = edge_color
+            self.refresh_edge_color()
+
+        else:
+            transformed_color = transform_color_with_defaults(
+                num_entries=len(self.data),
+                colors=edge_color,
+                elem_name="edge_color",
+                default="white",
+            )
+            self._edge_color = normalize_and_broadcast_colors(
+                len(self.data), transformed_color
+            )
+            self.edge_color_mode = ColorMode.DIRECT
+            self._edge_color_property = ''
+
+            self.events.edge_color()
+            self.events.highlight()
+
+    @property
+    def edge_color_cycle(self):
+        """Union[list, np.ndarray, cycle]:  Color cycle for edge_color. Can be a list of colors or a cycle of colors"""
+        return self._edge_color_cycle
+
+    @edge_color_cycle.setter
+    def edge_color_cycle(
+        self, edge_color_cycle: Union[list, np.ndarray, cycle]
+    ):
+        if isinstance(edge_color_cycle, cycle):
+            self._edge_color_cycle = edge_color_cycle
+        else:
+            self._edge_color_cycle = transform_color_cycle(
+                color_cycle=edge_color_cycle,
+                elem_name="edge_color_cycle",
+                default="white",
+            )
```

Comment:
Since this code is repeated, it would be nice to write a function for this (or, rather, just use this setter directly in the constructor!). If so, this setter probably needs to accept None (make the above type `Optional[]`).